### PR TITLE
dev-lang/zig: version bump to 0.7.1

### PR DESCRIPTION
**Pull request by epsilon-0**
also readds `shared` flag to `sys-devel/lld` as zig needs lld libraries.

### CODE REVIEW

**Review by juippis — APPROVED** [review]
I've just finished build-testing every ebuild in this PR and they work. I tried calling `zig` and it printed the help message, and also compiled `firefox` with the modified `lld`. It worked. So to me, it looks like this works, but I'm not knowledgeable enough to say for sure.

https://dev.gentoo.org/~juippis/logs/zig-gh-pr-19606.txt

**Review by mgorny — APPROVED** [review]
Feel free to merge.

**Comment by juippis** [comment]
And if needed, we can mask the new versions and run a tinderbox through revdeps.

File: dev-lang/zig/zig-0.7.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+ALL_LLVM_TARGETS=( "${ALL_LLVM_TARGETS[@]/#/llvm_targets_}" )
+LLVM_TARGET_USEDEPS="${ALL_LLVM_TARGETS[@]}"
+
+LLVM_MAX_SLOT=11
+
+RDEPEND="
+	sys-devel/clang:${LLVM_MAX_SLOT}
+	>=sys-devel/lld-11.0.0
+	<sys-devel/lld-12.0.0
```
Comment (mjsir911) [inline dev-lang/zig/zig-0.7.1.ebuild:36]: Not to be super annoying, but this doesn't actually work. A valid package (and the one resolving for me!) is `sys-devel/lld-11.0.1`, which hasn't been touched my this PR. Would a solution be to just backport the fix to existing ebuilds? unmasking the new revisions? adding a `shared` flag to existing ebuilds? I can file something if you give me a good idea and you want @mgorny

Comment (mjsir911) [reply dev-lang/zig/zig-0.7.1.ebuild:36]: See: https://bugs.gentoo.org/778398

**Pull request closed without merging**